api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImplTest.java

```diff
@@ -95,91 +118,201 @@ public void getLocationByUuid_shouldReturnNullWithUnknownUuid() {
-		Collection<Location> locations = fhirLocationDao.findLocationsByState(LOCATION_STATE);
+	public void searchForLocations_shouldReturnCorrectLocationByState() {
+		StringOrListParam state = new StringOrListParam().add(new StringParam(LOCATION_STATE));
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, state, null, null, null);
 		
 		assertThat(locations, notNullValue());
-		assertThat(locations.size(), greaterThanOrEqualTo(1));
+		assertThat(locations.size(), equalTo(2));
+		assertThat(locations.iterator().next().getStateProvince(), equalTo(LOCATION_STATE));
 	}
 	
 	@Test
-	public void findLocationsByState_shouldReturnEmptyCollectionWhenCalledWithUnknownState() {
-		Collection<Location> locations = fhirLocationDao.findLocationsByState(UNKNOWN_LOCATION_STATE);
+	public void searchForLocations_shouldReturnEmptyCollectionWhenCalledWithUnknownState() {
+		StringOrListParam state = new StringOrListParam().add(new StringParam(UNKNOWN_LOCATION_STATE));
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, state, null, null, null);
 		
 		assertThat(locations, notNullValue());
 		assertThat(locations.size(), equalTo(0));
 	}
 	
 	@Test
-	public void findLocationsByTag_shouldReturnLocationsContainingGivenTag() {
-		TokenParam locationTag = new TokenParam(FhirConstants.OPENMRS_FHIR_EXT_LOCATION_TAG, LOGIN_LOCATION_TAG_NAME);
-		Collection<Location> locations = fhirLocationDao.findLocationsByTag(locationTag);
+	public void searchForLocations_shouldReturnLocationsContainingGivenTag() {
+		TokenOrListParam locationTag = new TokenOrListParam(FhirConstants.OPENMRS_FHIR_EXT_LOCATION_TAG,
+		        LOGIN_LOCATION_TAG_NAME);
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, null, locationTag, null,
+		    null);
+		
+		assertThat(locations, notNullValue());
+		assertThat(locations.size(), equalTo(2));
+		assertThat(locations.iterator().next().getTags().iterator().next().getName(), equalTo(LOGIN_LOCATION_TAG_NAME));
+	}
+	
+	@Test
+	public void searchForLocations_shouldReturnCorrectLocationByParentUUID() {
+		ReferenceOrListParam parentLocation = new ReferenceOrListParam()
+		        .add(new ReferenceParam().setChain("").setValue(LOCATION_PARENT_ID));
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, null, null,
+		    parentLocation, null);
+		
 		assertThat(locations, notNullValue());
-		assertThat(locations.size(), greaterThanOrEqualTo(2));
+		assertThat(locations.size(), equalTo(1));
+		assertThat(locations.iterator().next().getParentLocation().getUuid(), equalTo(LOCATION_PARENT_ID));
+	}
+	
+	@Test
+	public void searchForLocations_shouldReturnCorrectLocationByParentName() {
+		ReferenceOrListParam parentLocation = new ReferenceOrListParam()
+		        .add(new ReferenceParam().setChain("name").setValue(LOCATION_PARENT_NAME));
```

Comment:
This should actually be `.setValue(...).setChain(...)` and so on for the others. The reason is that the `.setValue()` method resets the value of the chain. This will probably fix the coverage.

Reply:
This did fix the coverage! Thanks.